extension/apmconfigextension/opamp_callbacks.go

```diff
@@ -132,11 +132,14 @@ func (rc *remoteConfigCallbacks) onMessage(ctx context.Context, conn types.Conne
 		return &serverToAgent
 	}
 
-	if message.GetRemoteConfigStatus() != nil && message.GetRemoteConfigStatus().Status == protobufs.RemoteConfigStatuses_RemoteConfigStatuses_APPLIED && bytes.Equal(remoteConfig.ConfigHash, message.RemoteConfigStatus.GetLastRemoteConfigHash()) {
-		rc.logger.Info("Remote config applied", zap.String("hash", hex.EncodeToString(remoteConfig.ConfigHash)), agentUidField)
-		agent.lastConfigHash = message.GetRemoteConfigStatus().GetLastRemoteConfigHash()
+	remoteConfigStatus := message.GetRemoteConfigStatus()
```

Comment:
While at it maybe add a comment on our updated policy for updating the lastConfigHash?

Reply:
@xrmx I'm missing context. What policy do you mean?

Reply:
To resend the config only if the hash changed and not retry if the client failed to apply it

Reply:
Fixed in https://github.com/elastic/opentelemetry-collector-components/pull/609/commits/c277b97fd60663a6e5d009868c5efdd0395d48fe

Merging as discussed internally